lib/std/fs.zig

```diff
@@ -393,17 +393,20 @@ pub const Dir = struct {
                         self.index = 0;
                         self.end_index = @intCast(usize, rc);
                     }
-                    const freebsd_entry = @ptrCast(*align(1) os.dirent, &self.buf[self.index]);
-                    const next_index = self.index + freebsd_entry.reclen();
+                    const bsd_entry = @ptrCast(*align(1) os.dirent, &self.buf[self.index]);
+                    const next_index = self.index + bsd_entry.reclen();
                     self.index = next_index;
 
-                    const name = @ptrCast([*]u8, &freebsd_entry.d_name)[0..freebsd_entry.d_namlen];
+                    const name = @ptrCast([*]u8, &bsd_entry.d_name)[0..bsd_entry.d_namlen];
 
-                    if (mem.eql(u8, name, ".") or mem.eql(u8, name, "..")) {
+                    if (mem.eql(u8, name, ".") or mem.eql(u8, name, "..") or
+                            (builtin.os.tag == .openbsd and bsd_entry.d_fileno == 0) or
+                            (builtin.os.tag == .netbsd and bsd_entry.d_fileno == 0))
```

Comment:
I don't like too much this style. any alternative would be welcome

Reply:
```zig
const skip_zero_fileno = switch (builtin.os.tag) {
    // d_fileno=0 is used to mark invalid entries or deleted files.
    .openbsd, .netbsd => true,
    else => false,
};
if (mem.eql(u8, name, ".") or mem.eql(u8, name, "..") or
        (skip_zero_fileno and bsd_entry.d_fileno == 0))